packages/core/src/parseAmplifyGen2.ts

```diff
@@ -0,0 +1,245 @@
+import { OAuthProvider } from './libraryUtils';
+import { Gen2AuthProperties } from './singleton/gen2/types';
+import { Gen2Config, LegacyConfig, ResourcesConfig } from './singleton/types';
+
+export function isGen2Config(
+	resourceConfig: ResourcesConfig | LegacyConfig | Gen2Config,
+): resourceConfig is Gen2Config {
+	return !!(resourceConfig as Gen2Config).$id;
+}
+
+export function parseGen2Config(resourceConfig: Gen2Config): ResourcesConfig {
```

Comment:
I found it really confusing to name this param `resourceConfig` since it's so close to the `ResourcesConfig` type we're trying to map this **to**. I think we should be able to just keep it concise like we do with the parseAWSExports and refer to the incoming config as just `config`.
```suggestion
export function parseGen2Config(config: Gen2Config): ResourcesConfig {
```